### PR TITLE
fixing broken links, typos, etc per qodana

### PR DESCRIPTION
**Pull request by ghost**
I would do this myself, except I wasn't sure how. We need to add certain terms to the qodona dictionary so it doesn't keep flagging them:

uncollectible
Zuora
arrear
proration
autodetection
instance
Boleto
Xtensible

Qodana is also catching words that are in URLs and code (ex: sourcedir, userguide, etc.). Any way to stop that?

### CODE REVIEW

File: userguide/payment/includes/payment-flows.adoc [hunk 1]
```diff
@@ -32,9 +32,9 @@ The result of the API call will depend on the type of HPP integration:
 
 Note that after calling *buildFormDescriptor*, the payment may or may not exist in Kill Bill: this will depend on the plugin. If it does however, it will be most likely in a **PENDING** state.
 
-Here are a few common scenarii:
```
Comment (pierre) [inline userguide/payment/includes/payment-flows.adoc:35]: `scenarii` is incorrect? Or maybe just too pedantic?

File: userguide/payment/includes/payment-states.adoc [hunk 2]
```diff
@@ -8,7 +8,7 @@ The user can then submit additional requests for the same payment (capture, refu
 The payments have a state which will drive the set of possible subsequent operations available. For instance, it is possible to make a capture call against a payment in a state `AUTH_SUCCESS` but it is not possible to make such an operation against a payment in an `AUTH_ERRORED` state.
 The set of payment state transitions are configured in an https://github.com/killbill/killbill/blob/master/payment/src/main/resources/org/killbill/billing/payment/PaymentStates.xml[xml file] and the Kill Bill payment subsystem is in charge to enforce the transitions. The following diagram below shows all the possible transitions:
 
-graphviz::assets/payment_states.dot[format=svg,align=center]
```
Comment (pierre) [inline userguide/payment/includes/payment-states.adoc:11]: This is actually correct, but Qodana isn't smart enough to know it.

**Review by pierre — CHANGES_REQUESTED** [review]
Just one change required (graphviz stuff).

I've updated the dictionary (https://github.com/killbill/killbill-docs/pull/355/commits/76889078456c28c8e0ddd6b79f7782ccbf4925f0) and regenerated the report in my PR.

Regarding the false positives in urls and code, I need to read about how to configure it to ignore this.

File: userguide/payment/includes/payment-flows.adoc (hunk 1 above)
Comment (ghost) [reply userguide/payment/includes/payment-flows.adoc:35]: It is indeed pedantic; not commonly used anymore.

File: userguide/payment/includes/payment-states.adoc (hunk 2 above)
Comment (ghost) [reply userguide/payment/includes/payment-states.adoc:11]: Okay - But have you seen what it looks like? It looks weird.

https://docs.killbill.io/latest/userguide_payment.html#_payment_states

Comment (ghost) [reply userguide/payment/includes/payment-states.adoc:11]: @pierre - I've changed this filename back. I will double-check to make sure it looks all right after it's published. Currently it has very tall top/bottom margins and looks smooshed.

Comment (pierre) [reply userguide/payment/includes/payment-states.adoc:11]: 👍 Maybe the graphviz settings need to be tweaked...

Comment (ghost) [reply userguide/payment/includes/payment-states.adoc:11]: Yes, I'll look into it.

**Pull request merged**